crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -903,22 +905,35 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
             },
             Err(_) => None,
         };
-        if let Some(ref payment_method) = payment_data.payment_method_info {
-            payments::tokenization::update_connector_mandate_details_in_payment_method(
+        if let Some(payment_method_id) = &m_payment_method_id {
+            let payment_method = state
+                .store
+                .find_payment_method(payment_method_id)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
```

Comment:
Does this retrieval need to happen here? Can't we do it in `GetTrackers` or something?

Reply:
No , in `GetTrackers` for payments_confirm we don't have the  `payment_attempt` present while constructing  the `GenericMandateData` struct, as it's present in a tokio block . So instead of trying to construct the PaymentMethod Object in `GetTrackers` for all the needed operations, thought to construct it in the `PostUpdateTrackers` for just for the specific flows, i.e, Psync / CompleteAuthorize

Reply:
@Aprabhat19 let's do such retrievals in `get_trackers` only. We might end up doing multiple retrievals if this is the case.